contracts/UnderwriterToken.sol

```diff
@@ -27,8 +28,89 @@ import "@thesis/solidity-contracts/contracts/token/ERC20WithPermit.sol";
 ///         Anyone can submit this signature on the user's behalf by calling the
 ///         permit function, as specified in EIP2612 standard, paying gas fees,
 ///         and possibly performing other actions in the same transaction.
-contract UnderwriterToken is ERC20WithPermit {
+contract UnderwriterToken is ERC20WithPermit, Checkpoints {
+    /// @notice Replay protection nonces that are used to construct EIP-712
+    ///         signature provided to `delegateBySig` function.
+    mapping(address => uint256) public delegationNonce;
```

Comment:
Why we do not reuse `nonce` from `ERC20WithPermit`? I see we reuse it in T token implementation [here](https://github.com/threshold-network/solidity-contracts/blob/6664c738660f79de3add7fdff735fcb19d5165ad/contracts/token/T.sol#L61).